tests/test_contact_specific_functions.py

```diff
@@ -72,40 +116,32 @@ def test_prune_using_aabbs_rod_rod():
     "Testing function with analytically verified values"
 
     "Intersecting rod and rod"
-    "dummy inputs were generated using chatgpt"
-    rod_one_position_collection = np.array([[1, 2, 3], [4, 5, 6], [7, 8, 9]])
-    rod_one_radius_collection = np.array([1.0, 0.5])
-    rod_one_length_collection = np.array([2.0, 1.0])
-    rod_two_position_collection = np.array([[2, 3, 4], [5, 6, 7], [8, 9, 10]])
-    rod_two_radius_collection = np.array([0.8, 1.2])
-    rod_two_length_collection = np.array([1.5, 2.5])
+    """Since both the rods have same position, node's radius and lenght, they are overlapping/intersecting in 3D space."""
+    rod_one = MockRod()
+    rod_two = MockRod()
     assert (
         _prune_using_aabbs_rod_rod(
-            rod_one_position_collection,
-            rod_one_radius_collection,
-            rod_one_length_collection,
-            rod_two_position_collection,
-            rod_two_radius_collection,
-            rod_two_length_collection,
+            rod_one.position_collection,
+            rod_one.radius_collection,
+            rod_one.length_collection,
+            rod_two.position_collection,
+            rod_two.radius_collection,
+            rod_two.length_collection,
         )
         == 0
     )
 
     "Non - Intersecting rod and rod"
-    rod_one_position_collection = np.array([[1, 2, 3], [4, 5, 6], [7, 8, 9]])
-    rod_one_radius_collection = np.array([0.5, 0.3])
-    rod_one_length_collection = np.array([1, 2])
-    rod_two_position_collection = np.array([[15, 16, 17], [18, 19, 20], [21, 22, 23]])
-    rod_two_radius_collection = np.array([0.4, 0.2])
-    rod_two_length_collection = np.array([2, 3])
+    """Changing the position of rod_two in 3D space so the rod_one and rod_two don't overlap/intersect."""
+    rod_two.position_collection = np.array([[20, 21], [22, 23], [24, 25]])
```

Comment:
please adjust the size of this too as @armantekinalp said here too, also if you are making this change here in position collection you should also update rod two's lengths to match the new spacing between the nodes in position collection. Alternatively you could change the position collection to make the lengths stay the same.